requirements.txt

```diff
@@ -1,4 +1,4 @@
--e git+https://github.com/oda-hub/dispatcher-app.git@master#egg=cdci_data_analysis[test]
+-e git+https://github.com/oda-hub/dispatcher-app.git@get_query_products-error-message#egg=cdci_data_analysis[test]
```

Comment:
ah but no,  in there https://github.com/oda-hub/dispatcher-app/runs/4371363093?check_suite_focus=true  it is specifically the dispatcher currently being investigated. So in this new `Action` it is not necessary to make these branch dependencies.

This change is not needed.

Reply:
it keeps failing for this change that I applied here: [https://github.com/oda-hub/dispatcher-app/pull/291/commits/c554f35689b0d58c0a3d4a44d30baef50071c73a](https://github.com/oda-hub/dispatcher-app/pull/291/commits/c554f35689b0d58c0a3d4a44d30baef50071c73a)

having enabled that env var really is preventing certain tests to behave properly, I still don't understand why it's there, and only there

Reply:
Let's see there in dispatcher, not in the plugin.

Reply:
I mean, can we not really stick to modifying the dispatcher until the current plugin behavior is preserved?

Reply:
If tests pass without the variable it's fine. It's not set in the production.

Reply:
The variable was there since a long time and the tests were passing. So, by default, the problem would be in the very recent changes, and this PR.
It is suspicious that the variable with similar apparent intent has two names DISPATCHER_DEBUG_MODE and DISPATCHER_DEBUG. They might be combined.
By default, we would hope to minimize changes unless the changes are well understood.
If you want to figure it out - go ahead! Actually https://github.com/oda-hub/dispatcher-app/pull/291 may even be an appropriate place to do it: it deals with ways of propagating exception, similarly to the variable in some cases.
I could look too, but later.

Reply:
>
> The variable was there since a long time and the tests were passing. So, by default, the problem would be in the very recent changes, and this PR.

Exactly, by re-enabling the check with that variable (and of course reverting the change applied here) the tests here would pass.

>
> If you want to figure it out - go ahead! Actually oda-hub/dispatcher-app#291 may even be an appropriate place to do it: it deals with ways of propagating exception, similarly to the variable in some cases.

I will look into it and see what I discover, if my research is inconclusive we revert that change and and deal with that at a later stage, perhaps in [https://github.com/oda-hub/dispatcher-app/issues/273](https://github.com/oda-hub/dispatcher-app/issues/273).